src/Service/ServiceUtils.cpp

```diff
@@ -280,149 +278,116 @@ static ErrorMessageOr<fs::path> FindSymbolsFilePathInStructuredDebugStore(
       absl::StrFormat("File does not exist: %s", path_in_structured_debug_store.string())};
 }
 
-static ErrorMessageOr<fs::path> FindSymbolsFilePathCoff(const fs::path& module_path,
-                                                        const CoffFile* module_coff_file) {
-  CHECK(module_coff_file != nullptr);
+ErrorMessageOr<fs::path> FindSymbolsFilePath(
+    const orbit_grpc_protos::GetDebugInfoFileRequest& request) {
+  const fs::path module_path{request.module_path()};
 
-  // For Coff files, we currently only support debug symbols embedded in the object file.
-  if (module_coff_file->HasDebugSymbols()) {
-    return module_path;
+  // 1. Create object file for the module and check if it contains symbols itself.
+  auto object_file_or_error = CreateObjectFile(module_path);
+  if (object_file_or_error.has_error()) {
+    return ErrorMessage(absl::StrFormat("Unable to create object file: %s",
+                                        object_file_or_error.error().message()));
   }
-  std::string error_message_for_client{absl::StrFormat(
-      "Unable to find debug symbols on the instance for module \"%s\". ", module_path)};
-  return ErrorMessage(error_message_for_client);
-}
-
-static ErrorMessageOr<fs::path> FindSymbolsFilePathElf(
-    const fs::path& module_path, const ElfFile* module_elf_file,
-    const std::vector<fs::path>& search_directories) {
-  CHECK(module_elf_file != nullptr);
-
-  if (module_elf_file->HasDebugSymbols()) {
+  if (object_file_or_error.value()->HasDebugSymbols()) {
     return module_path;
   }
-
-  if (module_elf_file->GetBuildId().empty()) {
-    return ErrorMessage(absl::StrFormat(
-        "Unable to find symbols for module \"%s\". Module does not contain a build id",
-        module_path));
+  // 2. If module does not contain a build id, no searching for separate symbol files can be done
+  if (object_file_or_error.value()->GetBuildId().empty()) {
+    return ErrorMessage{
+        absl::StrFormat("Module \"%s\" does not contain symbols and does not contain a build id, "
+                        "therefore Orbit cannot search for a separate symbols file",
+                        module_path.string())};
   }
-
-  std::string build_id = module_elf_file->GetBuildId();
-
+  const std::string& build_id = object_file_or_error.value()->GetBuildId();
   if (build_id.size() < 3) {
```

Comment:
Isn't this check only correct in the ELF case?

Reply:
This build id check is actually related to the call to `FindSymbolsFilePathInStructuredDebugStore`, because this function assumes that the build id is long enough. I moved this `if` into the mentioned function and also disabled structured debug stores for coff files.

Reply:
Ok, thanks for clarifying!